### PR TITLE
Added script and secure variable for dependent builds

### PR DESCRIPTION
**Pull request by tldavies**

### CODE REVIEW

**Comment by tldavies** [comment]
@jlongever - Here is the PR for when on-core merges to master it reruns all the depends

File: build_children.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+curl -X POST https://api.travis-ci.org/builds/$BUILD_NUM_SYSLOG/restart --header "Authorization: token "$AUTH_TOKEN
+
+#Echo out what builds where restarted
+echo Restarted on-dhcp-proxy Build = $BUILD_NUM_DHCP
+echo Restarted on-tasks = $BUILD_NUM_TASKS
+echo Restarted on-tftp Build = $BUILD_NUM_TFTP
+echo Restarted on-syslog Build = $BUILD_NUM_SYSLOG
+
+
+echo "Note - on-core restarts builds for on-dhcp-proxy, on-tftp, on-syslog, and on-tasks. When on-tasks succeeds, it restarts builds for on-taskgraph and on-http. "
```
Comment (tldavies) [inline build_children.sh]: @benbp Does this work as a note?

File: build_children.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+#Gets the latest Master build
+GetMasterBuild () {
+
+    BUILD=$(curl -s 'https://api.travis-ci.org/repos/RackHD/'$1'/branches/master' | grep -o '{"branch":{"id":[0-9]*,' | grep -o '[0-9]' | tr -d '\n')
```
Comment (benbp) [inline build_children.sh]: I don't think you need or want quotes around `$1` here.

File: build_children.sh (hunk 1 above)
Comment (benbp) [inline build_children.sh]: Yes, thanks.

**Comment by benbp** [comment]
Do we end up running all the downstream builds 2x because of the node 4 and node 10 builds?

**Comment by jlongever** [comment]
Ignore that jenkins failure for now.

**Comment by tldavies** [comment]
@benbp Yes it tries to run them all twice cause of Node 4 and Node 10 builds. If it is already running the build though you get a fail error from the api. 

Error is build is already running: {"result":false,"flash":[{"error":"This build currently can not be restarted."}]}

**Comment by tldavies** [comment]
@RackHD/corecommitters

File: build_children.sh (hunk 2 above)
Comment (tldavies) [inline build_children.sh]: It didn;t work when there wasn't quotes.

**Comment by benbp** [comment]
@tldavies so if the node 4 build runs first, then we won't actually succeed in running the node 10 build at all?

**Comment by benbp** [comment]
Is there enough metadata from the API to grab both the most recent node 10 and node 4 builds and restart each individually?

**Comment by jlongever** [comment]
Question.. how does the on-core PR commit get applied to the downstream build?  I see the travis API calls to restart the last build for the dependencies but it's not clear how the on-core commit is applied to those builds.

**Comment by tldavies** [comment]
@benbp The Travis build runs both Node 4 and 10. So when it is restarted it starts both again. Node 4 vs Node 10 are different jobs but the same build. If we only ran the job than it would only run Node 10 or Node 4.

**Comment by tldavies** [comment]
@jlongever So this will only work when something is merged to on-core master. As forked PRs cannot access the environment variables. And then the dependent builds pull on-core from github which will have the newest version.

**Comment by jlongever** [comment]
I think the goal should be to prevent changes from being merged in that potentially break dependencies.

**Comment by benbp** [comment]
@jlongever yeah, it kind of sucks to not be able to do dependent PR builds before merge, but I think the tradeoff might be better than all the false negatives we've been getting from Jenkins at the same time.

**Comment by jlongever** [comment]
not really sure what the value is other than builds happening faster?   the consistency issues with jenkins are solvable.

**Comment by benbp** [comment]
If they're solvable, then yeah, we should probably just plan on migrating to Jenkins fully.

**Comment by benbp** [comment]
I do think builds happening fast has a high value though as an organization's commit velocity increases.

**Comment by benbp** [comment]
:+1:

**Comment by jlongever** [comment]
test this please

**Comment by VulpesArtificem** [comment]
:+1:

**Comment by tldavies** [comment]
test this please

**Comment by jlongever** [comment]
test this please

**Comment by tldavies** [comment]
test this please

**Comment by jlongever** [comment]
test this please

**Pull request merged**